File: FindMissing.java
// Time Complexity o(n) space complexity o(1);
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> missing = new ArrayList<Integer>();
        for(int i=0;i<nums.length;i++){
            int idx = Math.abs(nums[i])-1;
            if(nums[idx]>0){
                nums[idx]*=-1;
            }
        }
        for(int i=0;i<nums.length;i++){
            if(nums[i] >0){
                missing.add(i+1);
                
            }
            else{
                nums[i]*=-1;
            }
        }
        return missing;
    }
}

File: Find_missing.py
# Time Compelexity : O(n)
# Space Complexity: O(1)
# Approach to solve this problem:
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        missing = [] 
        
        for i in nums:
            pos = abs(i) - 1             
            if nums[pos] > 0:            # Everything is marked after this for loop
                nums[pos] *= -1                
        
        for i in range(len(nums)):       # What ever we find misssing we append
            if nums[i] > 0:
                missing.append(i+1)
                
        return missing